.github/workflows/lineageos-update.yaml

```diff
@@ -0,0 +1,37 @@
+name: "Update LineageOS"
+
+on: { schedule: [{ cron: '0 0 * * 0' }], workflow_dispatch }
+
+jobs:
+  updates:
+    name: "Update LineageOS"
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.3.5
+    - uses: cachix/install-nix-action@v14.1
+      with:
+        install_url: https://github.com/numtide/nix-unstable-installer/releases/download/nix-2.5pre20211019_4a2b7cc/install
+        extra_nix_config: "experimental-features = nix-command flakes"
+    - name: "Set OLD_BUILD_NUMBER variable"
+      run: echo "OLD_BUILD_NUMBER=$(nix eval -f . --arg configuration "{flavor = \"lineageos\";}" --raw config.buildNumber)" >> $GITHUB_ENV
+    - name: "Fetch latest lineage-18.1 changes"
+      run: nix develop -c ./flavors/lineageos/update.sh lineage-18.1
+    - name: "Fetch latest lineage-17.1 changes"
+      run: nix develop -c ./flavors/lineageos/update.sh lineage-17.1
+    - name: "Update buildDateTime"
+      run: sed -i "s/buildDateTime = mkDefault .*/buildDateTime = mkDefault $(date +%s);/" flavors/lineageos/default.nix
```

Comment:
It'd probably be simpler to put build time into a plain text file instead and import that in Nix.

Reply:
I stopped working on this PR because github actions storage is too little for this.